src/org/thoughtcrime/securesms/FullMsgActivity.java

```diff
@@ -0,0 +1,147 @@
+
+import java.lang.ref.WeakReference;
+
+import static android.util.Base64.*;
+
+public class FullMsgActivity extends WebViewActivity
+{
+  public static final String MSG_ID_EXTRA = "msg_id";
+  private int msgId;
+  private ApplicationDcContext dcContext;
+  private boolean loadRemoteContent = false;
+
+  enum LoadRemoteContent {
+    NEVER,
+    ONCE,
+    ALWAYS
+  };
+
+  @Override
+  protected void onCreate(Bundle state, boolean ready) {
+    super.onCreate(state, ready);
+
+    loadRemoteContent = Prefs.getAlwaysLoadRemoteContent(this);
+    webView.getSettings().setBlockNetworkLoads(!loadRemoteContent);
+
+    getSupportActionBar().setTitle(getString(R.string.chat_input_placeholder));
+
+    dcContext = DcHelper.getContext(this);
+    msgId = getIntent().getIntExtra(MSG_ID_EXTRA, 0);
+
+    new LoadHtmlAsyncTask(this).execute();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    if (isFinishing()) {
+      overridePendingTransition(R.anim.fade_scale_in, R.anim.slide_to_right);
+    }
+  }
+
+  @Override
+  public boolean onPrepareOptionsMenu(Menu menu) {
+    super.onPrepareOptionsMenu(menu);
+    this.getMenuInflater().inflate(R.menu.full_msg, menu);
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    super.onOptionsItemSelected(item);
+    switch (item.getItemId()) {
+      case R.id.load_remote_content:
+        AlertDialog.Builder builder = new AlertDialog.Builder(this)
+          .setTitle(R.string.load_remote_content)
+          .setMessage(R.string.load_remote_content_ask);
+
+        // we are using the buttons "[Always]  [Never][Once]" in that order.
+        // 1. Checkmarks before [Always] and [Never] show the current state.
+        // 2. [Once] is also shown in always-mode and disables always-mode if selected
+        //    (there was the idea to hide [Once] in always mode, but that looks more like a bug in the end)
+        // (maybe a usual Always-Checkbox and "[Cancel][OK]" buttons are an alternative, however, a [Once]
+        // would be required as well - probably as the leftmost button which is not that usable in
+        // not-always-mode where the dialog is used more often. Or [Ok] would mean "Once" as well as "Change checkbox setting",
+        // which is also a bit weird. Anyway, let's give the three buttons a try :)
+        if (Prefs.getAlwaysLoadRemoteContent(this)) {
+          builder.setNeutralButton("✔️ " + this.getString(R.string.always), (dialog, which) -> onChangeLoadRemoteContent(LoadRemoteContent.ALWAYS));
+          builder.setNegativeButton(R.string.never, (dialog, which) -> onChangeLoadRemoteContent(LoadRemoteContent.NEVER));
+        } else {
+          builder.setNeutralButton(R.string.always, (dialog, which) -> onChangeLoadRemoteContent(LoadRemoteContent.ALWAYS));
+          builder.setNegativeButton("✔️ " + this.getString(R.string.never), (dialog, which) -> onChangeLoadRemoteContent(LoadRemoteContent.NEVER));
```

Comment:
have you tested that on your device? can you post a screenshot? on android i am using here, these checkmarks are blue and contrast to dark as well as to light mode. but even black should not be that bad as the background is grey (and they are not so super-important anyway)

Reply:
![Screenshot_20210128-162052 (1)](https://user-images.githubusercontent.com/18012815/106159339-f772b800-6184-11eb-8d00-9969ae926f38.png)
![Screenshot_20210128-161915 (2)](https://user-images.githubusercontent.com/18012815/106159361-fb9ed580-6184-11eb-8678-7d096e1e0887.png)
(Fairphone 3, Android 10) 
Strange...

Reply:
k, thanks. another font or so. seems there are more fonts where that is black, https://emojipedia.org/check-mark/, so going for https://emojipedia.org/check-box-with-check/ in dark mode (as you suggested) seems to be reasonable.

btw. how does that toast when you forward a message to "Saved messages" look on your device? that is using the same checkmark.

Reply:
nevermind, i've added `DynamicTheme.getCheckmarkEmoji()` that is also use for that toast.

Reply:
The toast actually looks ok as toasts are always white-background on my device -> good contrast

Reply:
But now it looks good!

Reply:
+1, yeah, probably better to always use https://emojipedia.org/check-box-with-check/ in dark mode, in case the toast background is dark (because this checkmark always has good contrast)